internal/host/hostcommands/tang_connectivity_check_cmd.go

```diff
@@ -0,0 +1,51 @@
+package hostcommands
+
+import (
+	"context"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/host/hostutil"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type tangConnectivityCheckCmd struct {
+	baseCmd
+	db                         *gorm.DB
+	tangConnectivityCheckImage string
+}
+
+func NewTangConnectivityCheckCmd(log logrus.FieldLogger, db *gorm.DB, tangConnectivityCheckImage string) *tangConnectivityCheckCmd {
+	return &tangConnectivityCheckCmd{
+		baseCmd:                    baseCmd{log: log},
+		db:                         db,
+		tangConnectivityCheckImage: tangConnectivityCheckImage,
+	}
+}
+
+func (c *tangConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models.Step, error) {
+	var cluster common.Cluster
+
+	if err := c.db.First(&cluster, "id = ?", host.ClusterID).Error; err != nil {
+		c.log.WithError(err).Errorf("failed to fetch cluster %s", host.ClusterID)
+		return nil, err
+	}
+
+	if cluster.DiskEncryption == nil ||
+		swag.StringValue(cluster.DiskEncryption.EnableOn) == models.DiskEncryptionEnableOnNone ||
+		swag.StringValue(cluster.DiskEncryption.Mode) == models.DiskEncryptionModeTpmv2 ||
+		!hostutil.IsDiskEncryptionEnabledForRole(*cluster.DiskEncryption, common.GetEffectiveRole(host)) {
```

Comment:
It's a bit hard to follow this condition:) can you maybe split this to vars or add some comment. just to make it clearer.

Reply:
Ack.
How about:
```
	// Skip tangConnectivityCheck for cases where:
	// 1. DiskEncryption not set or not enabled.
	// 2. DiskEncryption mode is not tang based.
	// 3. DiskEncryption is not enabled, for the host role.
```
The last condition is the same helper function we currently use in `diskEncryptionRequirementsSatisfied` validation. In My cast I'm using it with a `not`.

Reply:
Yeah, should be clearer.

Reply:
Done.